src/target/stream.c

```diff
@@ -111,17 +163,44 @@ new_stream_target( FILE *stream ) {
 int
 sendto_stream_target( struct stream_target *target,
                       const char *msg,
-                      size_t msg_length ) {
+                      size_t msg_length,
+                      const struct stumpless_entry *entry ) {
   size_t fwrite_result;
+  enum stumpless_severity severity = stumpless_get_entry_severity(entry);
+  const char *sev_code = target->escape_codes[severity];
+  unsigned short sev_code_len = strlen(sev_code);
+  const char *reset_code = "\33[0m";
+  unsigned short reset_code_len = strlen(sev_code);
+  
+  if (sev_code_len != 0)
+  {
+    config_lock_mutex( &target->stream_mutex );
```

Comment:
you'll probably need to expand this from three individual critical sections to a single larger one, locking before the first write and only unlocking after the reset code is written. Otherwise, writes may end up being interleaved between messages with strange resulting output.

Reply:
Done.